src/webui.c

```diff
@@ -77,6 +77,33 @@ fetch_wa_cookie(struct evhttp_request *req)
     return wa;
 }
 
+static void
+send_json_worker(struct evhttp_request *req, void *arg)
+{
+    struct evbuffer *buf = evhttp_request_get_output_buffer(req);
+    struct evkeyvalq *hdrs_out = NULL;
+    double mh[6] = {0};
+    char rig_id[MAX_RIG_ID] = {0};
+    const char *url = evhttp_request_get_uri(req);
+    url += 8;
+    url = evhttp_uridecode(url, 1, NULL);
+    strncpy(rig_id, url, MAX_RIG_ID-1);
```

Comment:
It's still not quite there. 

If the URL *ends* with a slash, you'll want to drop that from the worker name section, and you'll want to do that *before* decoding, so you allow worker names *with* a slash encoded in them. Something like this:

```c
    char buf[128] = {0};
    char *wn = NULL;
    const char *url = evhttp_request_get_uri(req);

    url += 8; /* skip /worker/ */
    strncpy(buf, url, sizeof(buf)-1);
    buf[strcspn(buf, "/")] = 0;
    wn = evhttp_uridecode(buf, 1, NULL);
    if (!wn)
    {
        /* send error response */
    }
    strncpy(rig_id, wn, MAX_RIG_ID-1);
    free(wn);
    ...
```

That should cover:

```
/worker/aaa         -> aaa
/worker/bbb/        -> bbb
/worker/ccc/zzz...  -> ccc
/worker/ddd%2Feee   -> ddd/eee
/worker/fff%2Fggg/  -> fff/ggg
```

Reply:
The joys of handling user input on the web.